prow/tide/tide.go

```diff
@@ -902,13 +906,13 @@ func (c *Controller) takeAction(sp subpool, presubmits map[int]sets.String, batc
 		}
 	}
 	// If we have no serial jobs pending or successful, trigger one.
-	if len(nones) > 0 && len(pendings) == 0 && len(successes) == 0 {
+	if len(nones) > 0 && len(pendings) == 0 && len(successes) == 0 && !sp.headContextOnly {
```

Comment:
In the case where we have no jobs, we always take one of two actions -- either merge, or wait. Maybe more readable to make that explicit above instead of adding to these checks. Or make `c.trigger()` handle an empty list of presubmits correctly and forgo this logic entirely.